src/machine.rs

```diff
@@ -178,6 +178,20 @@ impl OlmMachine {
         future_to_promise(async move { Ok(me.display_name().await?) })
     }
 
+    /// Mark all tracked users as dirty.
+    ///
+    /// All users *whose device lists we are tracking* are flagged as needing a
+    /// key query. Users whose devices we are not tracking are ignored.
+    #[wasm_bindgen(js_name = "markAllTrackedUsersAsDirty")]
+    pub fn mark_all_tracked_users_as_dirty(&self) -> Promise {
+        let machine = self.inner.clone();
+
+        future_to_promise(async move {
+            machine.mark_all_tracked_users_as_dirty().await?;
+            Ok(JsValue::UNDEFINED)
+        })
+    }
```

Comment:
Turns out we don't need to call future_to_promise explicitly nowadays: we can just define an `async` fn and wasm-bindgen will do the work of turning the Rust Futures into JS Promises. We end up with cleaner code and better types in the generated TS.

I think this will work:

```suggestion
    pub async fn mark_all_tracked_users_as_dirty(&self) -> Result<(), JsError> {
        self.inner.mark_all_tracked_users_as_dirty().await?;
        Ok(())
    }
```